spec/std/object_spec.cr

```diff
@@ -333,4 +333,36 @@ describe Object do
       obj.property10?.should be_false
     end
   end
+
+  describe "try" do
+    it "yields self when not nil" do
+      obj = "An arbitrary string"
+      obj.try(&.itself).should eq(obj)
+    end
+
+    it "yields self when not nil even when default given" do
+      obj = "An arbitrary string"
+      obj.try("another value") {|o| o}.should eq(obj)
```

Comment:
yes, I did that idiomatic crystal in other examples, but wanted to show that this way resulted in the same behaviour